HadoopIllegalArgumentException occurred in the table with HDFS erasure coding

Hi,
We use prestosql 324 but ```HadoopIllegalArgumentException``` occurred in the table with HDFS erasure coding.

Any suggestion?

DDL
```
CREATE EXTERNAL TABLE hoge.piyo (
       hh STRING,
       mmss STRING,
       log STRING  COMMENT 'json'
)
PARTITIONED BY (service STRING, yyyymmdd STRING)
STORED AS RCFILE
LOCATION
'hdfs://...'
;
```

query
```
SELECT count(*) FROM hoge.piyo WHERE ... AND yyyymmdd='20190930' LIMIT 100
```

stacktrace
```
io.prestosql.spi.PrestoException: Failed to read RC file: hdfs://.../yyyymmdd=20190930/000044_0
	at io.prestosql.plugin.hive.rcfile.RcFilePageSource.getNextPage(RcFilePageSource.java:152)
	at io.prestosql.plugin.hive.HivePageSource.getNextPage(HivePageSource.java:244)
	at io.prestosql.operator.TableScanOperator.getOutput(TableScanOperator.java:286)
	at io.prestosql.operator.Driver.processInternal(Driver.java:379)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_324____20191204_075334_2.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.hadoop.HadoopIllegalArgumentException: Invalid buffer, not of length 951251
	at org.apache.hadoop.io.erasurecode.rawcoder.ByteBufferDecodingState.checkOutputBuffers(ByteBufferDecodingState.java:137)
	at org.apache.hadoop.io.erasurecode.rawcoder.ByteBufferDecodingState.<init>(ByteBufferDecodingState.java:48)
	at org.apache.hadoop.io.erasurecode.rawcoder.RawErasureDecoder.decode(RawErasureDecoder.java:86)
	at org.apache.hadoop.io.erasurecode.rawcoder.RawErasureDecoder.decode(RawErasureDecoder.java:170)
	at org.apache.hadoop.hdfs.StripeReader.decodeAndFillBuffer(StripeReader.java:433)
	at org.apache.hadoop.hdfs.PositionStripeReader.decode(PositionStripeReader.java:74)
	at org.apache.hadoop.hdfs.StripeReader.readStripe(StripeReader.java:390)
	at org.apache.hadoop.hdfs.DFSStripedInputStream.fetchBlockByteRange(DFSStripedInputStream.java:507)
	at org.apache.hadoop.hdfs.DFSInputStream.pread(DFSInputStream.java:1360)
	at org.apache.hadoop.hdfs.DFSInputStream.read(DFSInputStream.java:1324)
	at org.apache.hadoop.fs.FSInputStream.readFully(FSInputStream.java:121)
	at org.apache.hadoop.fs.FSDataInputStream.readFully(FSDataInputStream.java:111)
	at org.apache.hadoop.fs.FSDataInputStream.readFully(FSDataInputStream.java:111)
	at io.prestosql.plugin.hive.rcfile.HdfsRcFileDataSource.readFully(HdfsRcFileDataSource.java:82)
	at io.prestosql.rcfile.RcFileReader$DataSourceSliceLoader.load(RcFileReader.java:693)
	at io.prestosql.rcfile.RcFileReader$DataSourceSliceLoader.load(RcFileReader.java:667)
	at io.airlift.slice.ChunkedSliceInput$InternalLoader.load(ChunkedSliceInput.java:383)
	at io.airlift.slice.ChunkedSliceInput.ensureAvailable(ChunkedSliceInput.java:111)
	at io.airlift.slice.ChunkedSliceInput.readBytes(ChunkedSliceInput.java:259)
	at io.airlift.slice.ChunkedSliceInput.read(ChunkedSliceInput.java:242)
	at com.google.common.io.ByteStreams.skipUpTo(ByteStreams.java:819)
	at com.google.common.io.ByteStreams.skipFully(ByteStreams.java:795)
	at io.prestosql.rcfile.RcFileReader.advance(RcFileReader.java:410)
	at io.prestosql.plugin.hive.rcfile.RcFilePageSource.getNextPage(RcFilePageSource.java:122)
	... 13 more
```

reopened because it happend again.
error changed if I ran again although it's weird.

* pattern 1
```
io.prestosql.spi.PrestoException: Failed to read RC file: hdfs://.../yyyymmdd=20191231/000514_0
	at io.prestosql.plugin.hive.rcfile.RcFilePageSource.getNextPage(RcFilePageSource.java:152)
	at io.prestosql.plugin.hive.HivePageSource.getNextPage(HivePageSource.java:244)
	at io.prestosql.operator.TableScanOperator.getOutput(TableScanOperator.java:286)
	at io.prestosql.operator.Driver.processInternal(Driver.java:379)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_324____20191205_031231_2.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.hadoop.HadoopIllegalArgumentException: Invalid buffer, not of length 890036
	at org.apache.hadoop.io.erasurecode.rawcoder.ByteBufferDecodingState.checkOutputBuffers(ByteBufferDecodingState.java:137)
	at org.apache.hadoop.io.erasurecode.rawcoder.ByteBufferDecodingState.<init>(ByteBufferDecodingState.java:48)
	at org.apache.hadoop.io.erasurecode.rawcoder.RawErasureDecoder.decode(RawErasureDecoder.java:86)
	at org.apache.hadoop.io.erasurecode.rawcoder.RawErasureDecoder.decode(RawErasureDecoder.java:170)
	at org.apache.hadoop.hdfs.StripeReader.decodeAndFillBuffer(StripeReader.java:433)
	at org.apache.hadoop.hdfs.PositionStripeReader.decode(PositionStripeReader.java:74)
	at org.apache.hadoop.hdfs.StripeReader.readStripe(StripeReader.java:390)
	at org.apache.hadoop.hdfs.DFSStripedInputStream.fetchBlockByteRange(DFSStripedInputStream.java:507)
	at org.apache.hadoop.hdfs.DFSInputStream.pread(DFSInputStream.java:1360)
	at org.apache.hadoop.hdfs.DFSInputStream.read(DFSInputStream.java:1324)
	at org.apache.hadoop.fs.FSInputStream.readFully(FSInputStream.java:121)
	at org.apache.hadoop.fs.FSDataInputStream.readFully(FSDataInputStream.java:111)
	at org.apache.hadoop.fs.FSDataInputStream.readFully(FSDataInputStream.java:111)
	at io.prestosql.plugin.hive.rcfile.HdfsRcFileDataSource.readFully(HdfsRcFileDataSource.java:82)
	at io.prestosql.rcfile.RcFileReader$DataSourceSliceLoader.load(RcFileReader.java:693)
	at io.prestosql.rcfile.RcFileReader$DataSourceSliceLoader.load(RcFileReader.java:667)
	at io.airlift.slice.ChunkedSliceInput$InternalLoader.load(ChunkedSliceInput.java:383)
	at io.airlift.slice.ChunkedSliceInput.ensureAvailable(ChunkedSliceInput.java:111)
	at io.airlift.slice.ChunkedSliceInput.readBytes(ChunkedSliceInput.java:259)
	at io.airlift.slice.ChunkedSliceInput.read(ChunkedSliceInput.java:242)
	at com.google.common.io.ByteStreams.skipUpTo(ByteStreams.java:819)
	at com.google.common.io.ByteStreams.skipFully(ByteStreams.java:795)
	at io.prestosql.rcfile.RcFileReader.advance(RcFileReader.java:410)
	at io.prestosql.plugin.hive.rcfile.RcFilePageSource.getNextPage(RcFilePageSource.java:122)
	... 13 more
```
* pattern 2
```
io.prestosql.spi.PrestoException: Corrupted RC file: hdfs://.../yyyymmdd=20191231/001134_0
	at io.prestosql.plugin.hive.rcfile.RcFilePageSource.getNextPage(RcFilePageSource.java:148)
	at io.prestosql.plugin.hive.HivePageSource.getNextPage(HivePageSource.java:244)
	at io.prestosql.operator.TableScanOperator.getOutput(TableScanOperator.java:286)
	at io.prestosql.operator.Driver.processInternal(Driver.java:379)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_324____20191205_031235_2.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.prestosql.rcfile.RcFileCorruptionException: Invalid uncompressed row group length -498498544
	at io.prestosql.rcfile.RcFileReader.corrupt(RcFileReader.java:481)
	at io.prestosql.rcfile.RcFileReader.verify(RcFileReader.java:474)
	at io.prestosql.rcfile.RcFileReader.advance(RcFileReader.java:357)
	at io.prestosql.plugin.hive.rcfile.RcFilePageSource.getNextPage(RcFilePageSource.java:122)
	... 13 more
```
* pattern 3
```
io.prestosql.spi.PrestoException: Failed to read RC file: hdfs://.../yyyymmdd=20191231/001134_0
	at io.prestosql.plugin.hive.rcfile.RcFilePageSource.getNextPage(RcFilePageSource.java:152)
	at io.prestosql.plugin.hive.HivePageSource.getNextPage(HivePageSource.java:244)
	at io.prestosql.operator.TableScanOperator.getOutput(TableScanOperator.java:286)
	at io.prestosql.operator.Driver.processInternal(Driver.java:379)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_324____20191205_031231_2.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IndexOutOfBoundsException: end index (-1073818970) must not be negative
	at io.airlift.slice.Preconditions.checkPositionIndexes(Preconditions.java:81)
	at io.airlift.slice.Slice.checkIndexLength(Slice.java:1267)
	at io.airlift.slice.Slice.slice(Slice.java:838)
	at io.prestosql.rcfile.RcFileReader.advance(RcFileReader.java:373)
	at io.prestosql.plugin.hive.rcfile.RcFilePageSource.getNextPage(RcFilePageSource.java:122)
	... 13 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -772,7 +772,7 @@
             <dependency>
                 <groupId>io.trino.hadoop</groupId>
                 <artifactId>hadoop-apache</artifactId>
-                <version>3.2.0-18</version>
+                <version>3.3.5-1</version>
             </dependency>
 
             <dependency>
@@ -1897,7 +1897,7 @@
             <dependency>
                 <groupId>org.xerial.snappy</groupId>
                 <artifactId>snappy-java</artifactId>
-                <version>1.1.8.4</version>
+                <version>1.1.10.0</version>
                 <exclusions>
                     <exclusion>
                         <groupId>org.osgi</groupId>

